### DISCUSSION TITLE
How to access request headers when mocking?

### DISCUSSION THREAD

**Discussion by suneettipirneni** [nodejs/undici#1185] [Q&A] [answered]
Nock has a functionality where you can provide a function to access the request data like so:

```ts
.reply(200, function handler() {
	return { example: this.req.headers.example };
})
```

However I'm not sure how do the same thing with undici mocking. I've looked at [the docs](https://undici.nodejs.org/#/docs/best-practices/mocking-request), but still no luck. Any assistance would be greatly appreciated.

**Comment by p16** [comment, marked as answer]
Hi @suneettipirneni ,
this seems to be already supported.
if you try the [example in the doc](https://undici.nodejs.org/#/docs/best-practices/mocking-request) and add a function instead of an object in the `reply`

```javascript
mockPool.intercept({
  path: '/bank-transfer',
  method: 'POST',
  headers: {
    'X-TOKEN-SECRET': 'SuperSecretToken',
  },
  body: JSON.stringify({
    recepient: '1234567890',
    amount: '100'
  })
}).reply(200, (...all) => {
  console.log('all', all)
  return {
    message: 'transaction processed'
  }
})
```

the argument of that function will be the options given to the `intercept` function 

```
all [
  {
    method: 'POST',
    headers: { 'X-TOKEN-SECRET': 'SuperSecretToken' },
    body: '{"recepient":"1234567890","amount":"100"}',
    origin: 'http://localhost:3000',
    path: '/bank-transfer'
  }
]
```

is that what you were after?

**Reply by suneettipirneni** [reply]
Whoops! I should've marked this as answered since I made the PR to add this functionality 😂

**Reply by p16** [reply]
perfect, I did not realized it was you in the commits 😅

**Reply by p16** [reply]
@suneettipirneni looking at the doc there is no mention of this https://undici.nodejs.org/#/docs/best-practices/mocking-request , should I open a PR adding a small paragraph about it? or is it somewhere else I missed?

**Reply by p16** [reply]
Added this https://github.com/nodejs/undici/pull/1456
